shaders/src/shadowing.fs

```diff
@@ -360,6 +360,11 @@ highp float shadowVsm(const highp sampler2DArray shadowMap, const uint layer,
     highp vec2 moments = texture(shadowMap, vec3(shadowPosition.xy, layer)).xy;
     highp float depth = shadowPosition.z;
 
+    // depth must be clamped to support floating-point depth formats. This is to avoid comparing a
+    // value from the depth texture (which is never greater than 1.0) with a greater-than-one
+    // comparison value (which is possible with floating-point formats).
+    depth = min(depth, 1.0f);
```

Comment:
Note that a saturate (clamp(0, 1, x)) is free on some (all?) architecture. It might be worth using a clamp here. Not sure if min gives you that. Not a big deal either way.